Documentation/ceph-best-practices.md

```diff
@@ -0,0 +1,516 @@
+
+Choosing which option to use depends on your desired management strategy. There is no single
+strategy we would recommend over any other.
+
+### Other Ceph Daemons
+
+Placing the other Ceph daemons follows the same logic and methods as MONs and OSDs: MGR, MDS, RGW,
+NFS-Ganesha, and RBD mirroring daemons can all be placed as desired. We won't discuss these in
+details here since there are fewer strong recommendations for where these should be placed. For more
+information, see the [placement configuration settings](./ceph-cluster-crd.md#placement-configuration-settings].)
+
+
+## Hardware Requirements and Requests
+
+Kubernetes can watch the system resources available on nodes and can help schedule applications
+automatically. Kubernetes uses Resource Requests to do this. For Rook, we are notably concerned
+about Kubernetes' scheduling of Ceph daemons.
+
+Kubernetes has two Resource Request types: Requests and Limits. Requests govern scheduling, and
+Limits instruct Kubernetes to kill and restart application Pods when they are over-consuming given
+Limits.
+
+When there are Ceph hardware requirements, treat those requirements as Requests, not Limits. This is
+because all Ceph daemons are critical for storage, and it is best to never set Resource Limits for
+Ceph Pods. If Ceph Daemons are over-consuming Requests, there is likely a failure scenario
+happening. In a failure scenario, killing a daemon beyond a Limit is likely to make an already bad
+situation worse. This could create a “thundering herds” situation where failures synchronize and
+magnify.
+
+Generally, storage is given minimum resource guarantees, and other applications should be limited so
+as not to interfere. This guideline already applies to bare-metal storage deployments, not only for
+Kubernetes.
+
+As you read on, it is important to note that all recommendations can be affected by how Ceph daemons
+are configured. For example, any configuration regarding caching. Keep in mind that individual
+configurations are out of scope for this document.
+
+### Resource Requests - MON/MGR
+
+Resource Requests for MONs and MGRs are straightforward. MONs try to keep memory usage to around
+1GB; however, that can expand under failure scenarios. We recommend 4 GB RAM and 4 CPU cores.
+
+Recommendations for MGR nodes are harder to make, since enabling more modules means higher usage. We
+recommend starting with 2 GB RAM and 2 CPU cores for MGRs. It is a good idea to look at the actual
+usage for Deployments, and do not forget to consider usage during failure scenarios.
+
+MONs:
+ * Request 4 CPU cores
+ * Request 4GB RAM (2.5GB minimum)
+
+MGR:
+ * Memory will grow the more MGR modules are enabled
+ * Request 2 GB RAM and 2 CPU cores
+
+### Resource Requests - OSD CPU
+
+Recommendations and calculations for OSD CPU are straightforward.
+
+Hardware recommendations:
+ * 1 x 2GHz CPU Thread per spinner
+ * 2 x GHz CPU Thread per SSD
+ * 4 x GHz CPU Thread per NVMe
```

Comment:
The ceph perf team recommended splitting an nvme into multiple OSDs or else the nvme can't be saturated. Did your testing show that giving more resources was able to use the nvme sufficiently?